File: 프로그래머스/lv0/120891. 369게임/369게임.cs
using System;
using System.Linq;

public class Solution {
    public int solution(int order) {
        return order.ToString()
                    .Count(n => 
                           {
                               int num = (int)char.GetNumericValue(n);
                               return num != 0 && num % 3 == 0;
                           });
    }
}

File: 프로그래머스/0/120891. 369게임/369게임.cpp
#include <string>

using namespace std;

int solution(int order) {
    int answer = 0;
    
    while (order != 0)
    {
        int remainder = order % 10;
        if (remainder != 0 && remainder % 3 == 0)
            ++answer;
        
        order /= 10;
    }
    
    return answer;
}